SCHEMAS.sql

```diff
@@ -0,0 +1,304 @@
+"SCHEMA FOR BOOKMYSHOW"
+
+CREATE TABLE USERS(
+    USERID INT IDENTITY(1,1) PRIMARY KEY ,
+    NAME VARCHAR(50),
+    EMAIL VARCHAR(50) UNIQUE,
+    PHONENUMBER VARCHAR(13),
+    PASSWORD VARCHAR(12),
+    REGISTRATIONDATE DATETIME DEFAULT CURRENT_TIMESTAMP
+  );
+
+CREATE TABLE MOVIES(
+    MOVIEID INT IDENTITY(1,1) PRIMARY KEY,
+    MOVIENAME VARCHAR(200),
+    MOVIEGENRE VARCHAR(100),
+    DURATION TIME,
+    LANGUAGE VARCHAR(50),
+    RELEASEDATE DATE,
+    RATING DECIMAL(3,1),
+    DESCRIPTION TEXT
+);
+
+CREATE TABLE THEATRES(
+    THEATREID INT IDENTITY(1,1) PRIMARY KEY,
+    NAME VARCHAR(200),
+    CITY VARCHAR(100),
+    STATE VARCHAR(100),
+    ZIPCODE VARCHAR(30),
+    TOTALSCREENS INT,
+    PHONENUMBER VARCHAR(13)
+);
+
+CREATE TABLE SCREENS (
+    SCREENID INT IDENTITY(1,1) PRIMARY KEY ,
+    THEATREID INT,
+    SCREENNUMBER INT,
+    SEATINGCAPACITY INT,
+    FOREIGN KEY (THEATREID) REFERENCES THEATRES(THEATREID)
+);
+
+CREATE TABLE SHOW (
+    SHOWID INT IDENTITY(1,1) PRIMARY KEY ,
+    MOVIEID INT,
+    SCREENID INT,
+    SHOWDATETIME DATETIME,
+    AVAILABLESEATS INT,
+    FOREIGN KEY (MOVIEID) REFERENCES MOVIES(MOVIEID),
+    FOREIGN KEY (SCREENID) REFERENCES SCREENS(SCREENID)
+);
+
+CREATE TABLE BOOKINGS (
+    BOOKINGID INT IDENTITY(1,1) PRIMARY KEY ,
+    USERID INT,
+    SHOWTIMEID INT,
+    BOOKINGDATE DATETIME DEFAULT CURRENT_TIMESTAMP,
+    TOTALAMOUNT DECIMAL(10, 2),
+    FOREIGN KEY (USERID) REFERENCES USERS(USERID),
+    FOREIGN KEY (SHOWTIMEID) REFERENCES SHOW(SHOWID)
+);
+
+CREATE TABLE TICKETS (
+    TICKETID INT IDENTITY(1,1) PRIMARY KEY,
+    BOOKINGID INT,
+    SEATNUMBER VARCHAR(10),
```

Comment:
From where this data is derived

Reply:
Updated schema to derive data from seat table